### ISSUE TITLE
ActiveMQ Artemis Connection Factory creation fails in native image

### ISSUE THREAD

**Issue by krezovic** [spring-projects/spring-boot#42414]
See reproducer: https://github.com/krezovic/artemis-native-demo

```
./mvnw clean package -P native
./mvnw native:compile-no-fork
target/artemis-native-demo
```

Results in

```
Caused by: java.lang.NoSuchMethodException: org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory.<init>(java.lang.String)
        at java.base@21.0.4/java.lang.Class.checkMethod(DynamicHub.java:1078)
        at java.base@21.0.4/java.lang.Class.getConstructor0(DynamicHub.java:1241)
        at java.base@21.0.4/java.lang.Class.getConstructor(DynamicHub.java:2442)
        at org.springframework.boot.autoconfigure.jms.artemis.ArtemisConnectionFactoryFactory.newNativeConnectionFactory(ArtemisConnectionFactoryFactory.java:146)
        at org.springframework.boot.autoconfigure.jms.artemis.ArtemisConnectionFactoryFactory.createNativeConnectionFactory(ArtemisConnectionFactoryFactory.java:134)
        at org.springframework.boot.autoconfigure.jms.artemis.ArtemisConnectionFactoryFactory.doCreateConnectionFactory(ArtemisConnectionFactoryFactory.java:95)
        at org.springframework.boot.autoconfigure.jms.artemis.ArtemisConnectionFactoryFactory.createConnectionFactory(ArtemisConnectionFactoryFactory.java:67)
```

Short guess, reflection hints are missing for factory creation.

**Comment by philwebb** [comment]
I think we either need a hint or we need to change `ArtemisConnectionFactoryFactory.createConnectionFactory(...)` to take a `Function<String,ActiveMQConnectionFactory>`.

**Comment by krezovic** [comment]
This seems to be the first issue on Google when searching for Artemis, Spring and native-image so I'll just leave a comment that I have updated to latest snapshot of spring-boot 3.4, generated metadata for artemis using native agent and the application now works as expected.

The changes were added as a separate commit in the repo provided in the original comment.

Thanks for the prompt fix from spring boot side.
